table/tables/state_remote.go

```diff
@@ -276,3 +303,24 @@ func (r *mockStateRemoteData) LockForWrite(tid int64, now, ts uint64) (uint64, e
 	}
 	return 0, nil
 }
+
+func (r *mockStateRemoteData) renewLeaseForRead(tid int64, ts uint64) (bool, error) {
+	record, ok := r.data[tid]
+	if !ok {
+		record = &stateRecord{
+			lockLease:    ts,
+			oldReadLease: ts,
+			lockType:     CachedTableLockRead,
+		}
+		r.data[tid] = record
+		return true, nil
+	}
+	if record.lockType != CachedTableLockRead {
+		return false, errors.New("The read lock can be renewed only in the read lock state")
+	}
+	if record.lockLease > ts {
```

Comment:
record.lockLease <= ts ?

Reply:
OK, here you mean ts is the new lease ...
But the operation should check the old lease still hold, otherwise you can't renew!